app/instrumentation-tests/src/org/commcare/tests/BaseTest.java

```diff
@@ -0,0 +1,56 @@
+package org.commcare.tests;
+
+import android.Manifest;
+
+import androidx.test.espresso.intent.rule.IntentsTestRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.LargeTest;
+import androidx.test.rule.ActivityTestRule;
+import androidx.test.rule.GrantPermissionRule;
+import org.commcare.CommCareApplication;
+import org.commcare.activities.DispatchActivity;
+import org.commcare.activities.FormEntryActivity;
+import org.commcare.utils.Utility;
+import org.junit.Rule;
+import org.junit.runner.RunWith;
+
+import static androidx.test.espresso.Espresso.pressBackUnconditionally;
+
+/**
+ * @author $|-|!˅@M
+ */
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public abstract class BaseTest {
+
+    @Rule
+    public ActivityTestRule<DispatchActivity> activityTestRule = new ActivityTestRule<>(DispatchActivity.class);
+
+    @Rule
+    public IntentsTestRule<FormEntryActivity> intentsRule = new IntentsTestRule<>(FormEntryActivity.class);
```

Comment:
Yeah I thought that most of our intents are from the FormEntryActivity only so decided to put it here. But yeah makes sense to avoid having it in a superclass when not all subclasses need it.

Reply:
Now when I was updating it, I got to know that we need atleast 1 IntentsTestRule to be able to install the app offline. So keeping it here.

Reply:
How is a intent test rule with `FormEntryActivity` related to offline ccz install ?

Reply:
Ahh, I see it now. Actually we only need `IntentTestRule` with any activity, so will update the code. I used `ActivityTestRule` while starting to write tests and forgot to update that.